src/main/java/frc/robot/subsystems/intake/IntakeIOReal.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfigurator;
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+public class IntakeIOReal implements IntakeIO {
+
+    private final CANSparkMax spinMotor =
+            new CANSparkMax(Ports.Intake.INTAKE_FRONT_MOTOR, CANSparkLowLevel.MotorType.kBrushless);
+    private final CANSparkMax centerMotor =
+            new CANSparkMax(
+                    Ports.Intake.INTAKE_CENTER_MOTOR, CANSparkLowLevel.MotorType.kBrushless);
+    private final TalonFX angleMotor = new TalonFX(21);
+    private final TalonFXConfigurator angleConfigurator;
+    private final DutyCycleOut dutyCycleRequest = new DutyCycleOut(0);
+    private final PositionVoltage positionRequest = new PositionVoltage(0);
+    private TalonFXConfiguration angleConfiguration = new TalonFXConfiguration();
+
+    public IntakeIOReal() {
+        angleConfiguration.Feedback.SensorToMechanismRatio = IntakeConstants.ANGLE_GEAR_RATIO;
+
+        angleConfiguration.Feedback.RotorToSensorRatio = 1;
+
+        angleConfigurator = angleMotor.getConfigurator();
+
+        angleConfigurator.apply(angleConfiguration.Slot0);
+
+        spinMotor.restoreFactoryDefaults();
+        centerMotor.restoreFactoryDefaults();
+        spinMotor.setIdleMode(CANSparkMax.IdleMode.kCoast);
+        centerMotor.setIdleMode(CANSparkMax.IdleMode.kCoast);
+        spinMotor.enableVoltageCompensation(Constants.NOMINAL_VOLTAGE);
+        centerMotor.enableVoltageCompensation(Constants.NOMINAL_VOLTAGE);
+        spinMotor.setInverted(Ports.Intake.POWER_INVERTED);
+        centerMotor.setInverted(Ports.Intake.POWER_INVERTED);
```

Comment:
All configurations with a `.with*()` method should be stored in the constants file.

Reply:
not needed

Reply:
ff661168fd68a446698795e33cafd3f79c0de43b

Reply:
ff661168fd68a446698795e33cafd3f79c0de43b

Reply:
This has not been edited.

Reply:
true but this is a neo spark max motor not a phoenix motor, so chaining configurations is not possible.